service/src/main/kotlin/fi/espoo/evaka/application/ApplicationStateService.kt

```diff
@@ -558,20 +506,20 @@ class ApplicationStateService(
             throw BadRequest("Expected status to be one of [${statuses.joinToString(separator = ", ")}] but was ${application.status}")
     }
 
-    private fun validateApplication(tx: Database.Read, application: ApplicationDetails) {
+    private fun validateApplication(tx: Database.Read, type: ApplicationType, application: ApplicationForm, strict: Boolean) {
         val result = ValidationResult()
 
-        val preferredStartDate = application.form.preferences.preferredStartDate
-        if (application.type == ApplicationType.PRESCHOOL && preferredStartDate != null) {
+        val preferredStartDate = application.preferences.preferredStartDate
+        if (type == ApplicationType.PRESCHOOL && preferredStartDate != null) {
```

Comment:
Small style suggestions: extract each validation as own function, for example fun validatePreschoolPreferredStartDate(...).

Reply:
I think this function needs refactoring when we do frontend validation, so I'll try to remember this then